jcl/src/openj9.sharedclasses/share/classes/com/ibm/oti/shared/SharedClassURLHelperImpl.java

```diff
@@ -47,12 +52,36 @@
 		initializeShareableClassloader(loader);
 	}
 	
-	private static native void init(); 
+	private static native void init();
+	
+	/* The passed-in URL path contains a jar file */
+	private boolean newJarFileCheck(URL convertedjarPath) {
+		String protocol = convertedjarPath.getProtocol();
+		String jarFile = convertedjarPath.getFile();
+		boolean isJarType = false;
+		
+		if (protocol.equalsIgnoreCase("jar")) { //$NON-NLS-1$
+			isJarType = true;
+		} else if (protocol.equalsIgnoreCase("file")) { //$NON-NLS-1$
+			if (jarFile.endsWith(".jar") || jarFile.endsWith(".zip") //$NON-NLS-1$ //$NON-NLS-2$
+			|| jarFile.contains("!/") || jarFile.contains("!\\") //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
@hangshao0 is this check for `!\` required? You added it to the native code, but not sure why.

Reply:
Yes, it is required. For nested jar.